e2e/test_operator/operator_test.go

```diff
@@ -142,6 +144,79 @@ var _ = Describe("Operator", Label("e2e"), func() {
 		})
 	})
 
+	When("taint a Node that hosts a Pod", func() {
+		var replacedPod corev1.Pod
+		taintKeyMaintenance := "maintenance"
+		taintKeyMaintenanceDuration := int64(2)
+		taintKeyStar := "*"
+		taintKeyStarDuration := int64(5)
+
+		BeforeEach(func() {
+			initialPods := fdbCluster.GetStatelessPods()
+			replacedPod = fixtures.RandomPickOnePod(initialPods.Items)
+			replacedPodProcessGroupID := internal.GetProcessGroupIDFromMeta(fdbCluster.GetCluster(), replacedPod.ObjectMeta)
+			// Taint replacePod's node
+			// Q: Why do I have to GetNode before UpdateNode()? otherwise, I got invalid node.spec.id error
+			// node := &corev1.Node{
+			// 	ObjectMeta: metav1.ObjectMeta{Name: replacedPod.Spec.NodeName},
+			// }
+			node := fdbCluster.GetNode(replacedPod.Spec.NodeName)
+			node.Spec.Taints = []corev1.Taint{
+				{
+					Key:       taintKeyMaintenance,
+					Value:     "rack_maintenance",
+					Effect:    corev1.TaintEffectNoExecute,
+					TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Second * time.Duration(taintKeyMaintenanceDuration+1))},
+				},
+			}
+			log.Printf("Taint node: Pod name:%s Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", replacedPod.Name,
+				node.Name, node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+			err := fdbCluster.UpdateNode(node)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Setup cluster's taint config and wait for long enough
+			curClusterSpec := fdbCluster.GetCluster().Spec.DeepCopy()
+			curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions = []fdbv1beta2.TaintReplacementOption{
+				{
+					Key:               &taintKeyStar,
+					DurationInSeconds: &taintKeyStarDuration,
+				},
+				{
+					Key:               &taintKeyMaintenance,
+					DurationInSeconds: &taintKeyMaintenanceDuration,
+				},
+			}
+			curClusterSpec.AutomationOptions.Replacements.TaintReplacementTimeSeconds = pointer.Int(1)
+			fdbCluster.UpdateClusterSpecWithSpec(curClusterSpec)
+
+			curClusterSpec = fdbCluster.GetCluster().Spec.DeepCopy()
+			if len(curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions) < 1 {
+				log.Printf("Update Cluster Taint Option failed %+v\n", curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions)
+			}
+			//Expect(*curClusterSpec.AutomationOptions.Replacements.TaintReplacementOptions[0]).To(taintKeyStar)
+
+			Expect(*fdbCluster.GetAutomationOptions().Replacements.Enabled).To(BeTrue())
+			processGroupStatus := fdbv1beta2.FindProcessGroupByID(fdbCluster.GetCluster().Status.ProcessGroups, replacedPodProcessGroupID)
+			Expect(processGroupStatus).NotTo(BeNil())
+			// TODO: reenable below
+			// Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintDetected)).NotTo(BeNil())
+			// Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintReplacing)).NotTo(BeNil())
+
+			// Wait for operator to replace the pod
```

Comment:
Shouldn't we check that the operator will replace all Pods on this node? There could be the case that more than one Pod is running on the same node.